documentation/release_notes.markdown

```diff
@@ -4,116 +4,104 @@ title: "Puppet Server: Release Notes"
-[resource type API]: https://docs.puppet.com/puppet/latest/reference/http_api/http_resource_type.html
+[Trapperkeeper]: https://github.com/puppetlabs/trapperkeeper
+[service bootstrapping]: ./configuration.markdown#service-bootstrapping
+[auth.conf]: ./config_file_auth.markdown
 
-## Puppet Server 2.3.2
+## Puppet Server 2.4
 
-Released April 27, 2016.
+Released May 19, 2016.
 
-This is a bug-fix release that also resolves a security issue in endpoint URL decoding.
+This is a feature and bug-fix release of Puppet Server that also upgrades its included [Trapperkeeper][] framework from version 1.3.1 to 1.4.0.
 
-### **Security fix:** CVE-2016-2785 - Correctly decode endpoint URLs
+This release also adds packages for Ubuntu 16.04 LTS (Xenial Xerus) and no longer includes packages for Fedora 21, which reached its end of life in December.
 
-Previous versions of Puppet Server 2.x did not correctly decode specific character combinations that could potentially allow for a host to access endpoints restricted by [`auth.conf`](./config_file_auth.markdown) rules. Puppet Server 2.3.2 resolves this issue.
+### New platform: Ubuntu 16.04 LTS (Xenial Xerus)
 
-* [CVE-2016-2785](https://puppet.com/security/cve/cve-2016-2785)
+Puppet Server 2.4.0 introduces [Puppet-built packages](https://docs.puppet.com/puppetserver/latest/install_from_packages.html) for Ubuntu 16.04 LTS (Xenial Xerus). For details about Puppet's package repositories, see the [Puppet Collections documentation](https://docs.puppet.com/puppet/latest/reference/puppet_collections.html).
 
-### Bug fix: Support Ruby master's custom trusted OID mappings
+-   [SERVER-1182](https://tickets.puppetlabs.com/browse/SERVER-1182)
 
-Puppet's Ruby master supports [custom trusted certificate extension object identifier (OID) mappings](https://docs.puppet.com/puppet/latest/reference/config_file_oid_map.html), but previous versions of Puppet Server 2.x did not. This could result in Puppet Server storing trusted extensions with their default numeric OIDs as their keys, instead of the expected custom-mapped short names.
+### New feature: Integrate with systemd services on Debian and Ubuntu
 
-Puppet Server 2.3.2 resolves this issue by supporting the `trusted_oid_mapping_file` and `csr_attributes` settings in [`puppet.conf`](https://docs.puppet.com/puppet/4.4/reference/config_file_main.html), which define paths to YAML files containing the custom mappings and certificate extension request (CSR) attributes, respectively.
+Puppet Server 2.4.0 adds integration with `systemd` on Debian 8 and newer, and Ubuntu 15.04 and newer.
 
-For more information about mapping custom trusted OID names, see the Puppet documentation for the  [`custom_trusted_oid_mapping.yaml` file](https://docs.puppet.com/puppet/latest/reference/config_file_oid_map.html).
+-   [EZ-48: Add systemd services for debian 8 and ubuntu >= 15.04](https://tickets.puppetlabs.com/browse/EZ-48)
```

Comment:
Also, systemd support ended up being *just* 16.04. there is follow-up work slated to move jessie and wily to systemd from sysv.